sql/api/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseParser.g4

```diff
@@ -61,11 +61,18 @@ compoundBody
 
 compoundStatement
     : statement
+    | compoundSetStatement
     | beginEndCompoundBlock
     ;
 
+compoundSetStatement
```

Comment:
why do we call it `compound`?

Reply:
we should name the parser rule based on what it matches, instead of who invokes it. How about `setVariableWithOptionalKeyword`?

Reply:
sure, will do shortly